recipes/mamba-ssm/recipe.yaml

```
@@ -0,0 +1,58 @@
+context:
+  name: mamba-ssm
+  version: 2.2.4
+
+package:
+  name: ${{ name|lower }}
+  version: ${{ version }}
+
+source:
+  url: https://pypi.org/packages/source/${{ name[0] }}/${{ name }}/mamba_ssm-${{ version }}.tar.gz
+  sha256: e4114c69302796c91b71e90032c2d974f611608fab331582a80de6eaf075efb9
+
+build:
+  number: 0
+  script: ${{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip:
+    # mamba-ssm requires CUDA (or ROCm) GPU
+    - cuda_compiler_version == "None"
+    # CUDA < 12 not supported by pytorch anymore
+    - cuda_compiler_version == "11.8"
+
+requirements:
```

```suggestion
requirements:
  build:
    - {{ compiler('c') }}
    - {{ stdlib('c') }}
    - {{ compiler('cxx') }}
    - {{ compiler('cuda') }}
```
Since this package appears have CUDA code in https://github.com/state-spaces/mamba/tree/main/csrc/selective_scan, it definitely needs at a minimum these build dependencies, and well as some others in `host`.

---

@jakirkham or @carterbox, could you suggest if a rebuild from source is necessary here, or can we just reuse the `.so` file from the PyPI wheel?

---

Binaries should always be built from source.  This is important for both licensing and dependency tracking.